feedback-rep2004.md

```diff
@@ -0,0 +1,44 @@
+We think adding a full 'report card' could be useful to assess how a package is failing to meet criteria for the higher quality level. For this 'report card' to be useful, it should be quickly and easily readable. We could consider a 'spider chart' (or [radar chart as per wikipedia](https://en.wikipedia.org/wiki/Radar_chart)), or a table like [here](https://github.com/ros-infrastructure/rep/blob/c8bd456f531acc4863d0bb8888c28b10f9271492/rep-2004.rst#quality-level-comparison-chart).
+
+### Security specific context and feedback
+
+#### Quality Levels and the Vulnerability Disclosure Policy
+The VDP tells the public we’d like a flat 90-day grace period between when we receive a vulnerability before it’s made public.  In the interest of keeping the VDP simple, I recommend keeping code quality out of this public-facing document.  We can--and should--strive to do better for high quality code, but I don’t believe that needs to be a part of this document.
+
+#### Quality Levels and Vulnerability Remediation
+Internally, however, we should set a higher standard for higher quality code.
+Level 1 and 2 code will be widely deployed in the field. Fixes should be made available this code quickly to protect production robots.
+Level 3 and some level 4 code may not be deployed in robots, but will be used daily by firms that build robots.  High risk vulnerabilities should be handled quickly to protect development environments.
+CVSS is the defacto standard for risk assessing vulnerabilities.  Although flawed, it’s better than trying to create a new standard.
+
+#### Remediation Timeline
+Here’s an initial proposal on responsiveness based on the principles above.  This would be a commitment by the ROS 2 developer community.
+Triage within 3 days for all reports, triage assigns a CVSS score and a maintainer to provide a fix
+Maintainer is expected to provide a fix according to this timeline; the maintainer may also change the CVSS score as they dig into the issue:
+
+
+|                   | High Risk<br> CVSS 9+ | Medium Risk<br> CVSS 6-9            | Low Risk<br>CVSS <6                       |
+| ----------------- | --------- | ---------------------- | ------------------------------ |
+| Quality Level 1   | 7 days    | 30 days                | No timeline, use issue         |
+| Quality Level 2   | 7 days    | 30 days                | No timeline, use issue         |
+| Quality Level 3   | 30 days   | No timeline, use issue | No timeline, use issue         |
+| Quality Level 4-5 | 30 days   | No timeline, use issue | No timeline, use issue         |
+
+
+
+Once the maintainer has approved the code update, the code is built and distributed according to the normal timeline.
+
+
+## Suggestions outside of REP but linked to it:
+
+- Consider providing tooling to help people evaluate / generate the quality level and files associated with quality level (could go from a simple list of Y/N questions to actually pulling info of all the dependencies etc)
```

Comment:
What exactly do you have in mind here, something like https://github.com/aliasrobotics/RSF particularized for ROS software development? We should probably be a bit more specific.

Reply:
We were (@artivis to confirm as it was his suggestion) hoping to reduce the amount of work to achieve the quality declaration expected for various packages.
The rep provides a list of criteria packages need to comply with to achieve a given quality level.
It looks like it would be a pain to write all those https://github.com/ros2/rcl/pull/566/files by hand. And providing a simpler way to generate those could be useful.
A tool will likely never cover specific cases but could help for e.g. verifying that all your dependencies are at a sufficient quality level for your package to use them.

While having methodologies similar to RSF tailored to ROS is definitely a great thing in the long term, it seems hard to fit in this REP that is (purposefully) vague.

Reply:
The overall idea is to provide small tools to help both a developer to estimate its package quality level and package consumers to assert that the quality level claimed is true. The former could be a simple CLI-based y/n form while the later would probably be more difficult to set up given the wide range of stuff covered by the quality level...
The rational is that because the 'quality level' is declarative (claimative?) but there are so many criterions, we need tools to ease our life (as usual). Not everyone will go to the REP in search for the criterions list, however we can reasonably expect people to run, say, `ros pkg eval_quality` and answer a bunch of y/n.
If we do have to be specific, I would only mention the package developer tool (the CLI-based y/n form) given that the second is likely not to exists anytime soon.

Reply:
> If we do have to be specific, I would only mention the package developer tool (the CLI-based y/n form) given that the second is likely not to exists anytime soon.

I don't think that we do have to, it wont make it in the REP (hence the "outside of REP" section) this is just to spark discussion on what type of tooling could make this REP easier to adopt.
As of now it's hard to evaluate the usefulness of a specific suggested tool without going through the effort writing the quality declaration of a package.
If taking into account automation allows to steer the format of the quality declaration files it would be a plus